src/peer/datastore.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+const { clone } = require('lodash')
+const Levelup = require('levelup')
+const serialize = require('../metadata/serialize')
+const Multihashing = require('multihashing')
+const Multihash = require('multihashes')
+
+export type DatastoreOptions = {
+  backend: 'memory', // just in-memory for now, expand to e.g. rocksdb
+  location: string
+}
+
+const DefaultOptions: DatastoreOptions = {
+  backend: 'memory',
+  location: '/aleph/data'
+}
+
+class Datastore {
+  db: Levelup
+
+  constructor (options: ?DatastoreOptions = null) {
+    if (options == null) {
```

Comment:
make the members optional? I'm actually not sure what `locaiton` even means for `memory` backend